File: 485-max-consecutive-ones/max-consecutive-ones.go
func findMaxConsecutiveOnes(nums []int) int {
    count := 0
    maxCount := 0

    for _, num := range nums {
        if num == 1 {
            count++
        } else {
            if count > maxCount {
                maxCount = count
            }
            count = 0
        }
    }

    if count > maxCount {
        maxCount = count
    }

    return maxCount
}

File: 485-max-consecutive-ones/max-consecutive-ones.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        count = 0
        maxCount = 0

        for num in nums:
            if num == 0:
                count = 0
            else:
                count += 1

            if maxCount < count:
                maxCount = count

        return maxCount
        